### PR TITLE
S3 publishing

### PR DESCRIPTION
**Pull request by jchoca**
This will publish our modules to S3, so we can make this repo private. We'll just need to update all the URLs to be of the format:

```terraform
module "statistics_image" {
  source    = "s3::https://terraform-building-blocks.s3.us-east-2.amazonaws.com/{BRANCH_OR_TAG}/ecr-url.zip"
  ecr_image = "postmark-statistics:${local.container_deploy_tag}"
}
```

### CODE REVIEW

File: scripts/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  for_each   = toset(local.source_folders)
+  bucket     = aws_s3_bucket.building_blocks.bucket
+  key        = "${var.git_ref}/${each.value}.zip"
+  source     = "${path.module}/../../dist/${each.value}.zip"
+  depends_on = [data.archive_file.module_packages]
+}
+
+locals {
+  ci_account_id  = 602216863712
+  source_folders = [
```
Comment (jchoca) [inline scripts/terraform/main.tf:87]: Maybe there is a sexier way to do this? :)

Ideally for `archive_file` we would use a file glob and then exclude certain folders, but it didn't seem to support that.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+version: 2.1
+
+jobs:
+  run-terraform:
+    parameters:
+      extra_args:
+        type: string
+        default: ''
+    docker:
+      - image: 602216863712.dkr.ecr.us-east-2.amazonaws.com/deploy-tools:tf-updates-alpine
```
Comment (jchoca) [inline .circleci/config.yml:10]: I'll update this once https://github.com/wildbit/postmark-containers/pull/29 is approved

File: scripts/terraform/main.tf (hunk 1 above)
Comment (Archanian) [reply scripts/terraform/main.tf:86]: Yeah I think it's ok. Also is `benthos` missing from this list intentionally?

Comment (jchoca) [reply scripts/terraform/main.tf:86]: Nope let me get that added

**Pull request merged**